src/client/Microsoft.Identity.Client/AuthenticationResultMetadata.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Contains metadata of the authentication result.
+    /// </summary>
+    public class AuthenticationResultMetadata
+    {
+        /// <summary>
+        /// Default constructor for the class AuthenticationResultMetadata
+        /// </summary>
+        public AuthenticationResultMetadata()
```

Comment:
I don't see any reason for this object to be mutable (i.e. have a default ctor and a property setter). Immutable objects are  much easier to work with - see https://www.linkedin.com/pulse/20140528113353-16837833-6-benefits-of-programming-with-immutable-objects-in-java/ ... Mutability for this particular object is not very important, but it's good practice to be immutable by default.